File: projekt2/src/proj/java/model/animal/Fox.java
package proj.java.model.animal;

import java.awt.Color;
import java.util.Random;

import proj.java.model.Animal;
import proj.java.model.World;

public class Fox extends Animal {

    private static final int FOX_STRENGTH = 3;
    private static final int FOX_INITIATIVE = 7;

    @Override
    public Color getColor() {
        return Color.ORANGE;
    }

    @Override
    public int getInitiative() {
        return FOX_INITIATIVE;
    }

    @Override
    public int getStrength() {
        return FOX_STRENGTH + getExtraStrength();
    }

    @Override
    protected void reproduce(World world, int row, int col) {
        world.setOrganism(row, col, new Fox());
    }

    @Override
    public void takeAction(World world) {
        var rand = new Random().nextInt(4);
        switch (rand) {
            case 0: // gora
                if (world.isInsideWorld(getRow() - 1, getCol())) {
                    if (world.getOrganism(getRow() - 1, getCol()) != null) {
                        if (world.getOrganism(getRow() - 1, getCol()).getExtraStrength() < this.getStrength())
                            move(getRow(), getCol(), getRow() - 1, getCol(), world);
                    } else
                        move(getRow(), getCol(), getRow() - 1, getCol(), world);
                }
                break;
            case 1: // prawo
                if (world.isInsideWorld(getRow(), getCol() + 1)) {
                    if (world.getOrganism(getRow(), getCol() + 1) != null) {
                        if (world.getOrganism(getRow(), getCol() + 1).getExtraStrength() < this.getStrength())
                            move(getRow(), getCol(), getRow(), getCol() + 1, world);
                    } else
                        move(getRow(), getCol(), getRow(), getCol() + 1, world);
                }
                break;
            case 2: // dol
                if (world.isInsideWorld(getRow() + 1, getCol())) {
                    if (world.getOrganism(getRow() + 1, getCol()) != null) {
                        if (world.getOrganism(getRow() + 1, getCol()).getExtraStrength() < this.getStrength())
                            move(getRow(), getCol(), getRow() + 1, getCol(), world);
                    } else
                        move(getRow(), getCol(), getRow() + 1, getCol(), world);
                }
                break;
            case 3: // lewo
                if (world.isInsideWorld(getRow(), getCol() - 1)) {
                    if (world.getOrganism(getRow(), getCol() - 1) != null) {
                        if (world.getOrganism(getRow(), getCol() - 1).getExtraStrength() < this.getStrength())
                            move(getRow(), getCol(), getRow(), getCol() - 1, world);
                    } else
                        move(getRow(), getCol(), getRow(), getCol() - 1, world);
                }
                break;
            default:
                break;
        }
    }

}

File: projekt3/Fox.py
from Animal import *

class Fox(Animal):
    __FOX_STRENGTH__ = 3
    __FOX_INITIATIVE__ = 7
    __COLOR__ = "Orange"
    
    def __init__(self, row, col, world):
        super().__init__(row, col, self.__FOX_STRENGTH__, self.__FOX_INITIATIVE__, world)
    
    def getColor(self):
        return self.__COLOR__
    
    def getName(self):
        return "Fox"

    def reproduce(self, row, col):
        self._world_.setOrganism(row, col, Fox())

    def takeAction(self):
        rand = randrange(4)
        if rand == 0: #gura
            if (self._world_.isInsideWorld(self.getRow() - 1, self.getCol())):
                    if (self._world_.getOrgFromPos(self.getRow() - 1, self.getCol()) != None):
                        other = self._world_.getOrgFromPos(self.getRow() - 1, self.getCol())
                        if (other.getStrength() < self.getStrength()):
                            self.move(self.getRow() - 1, self.getCol())
                    else:
                        self.move(self.getRow() - 1, self.getCol())
        elif rand == 1: #prawo
            if (self._world_.isInsideWorld(self.getRow(), self.getCol() + 1)):
                    if (self._world_.getOrgFromPos(self.getRow(), self.getCol() + 1) != None):
                        other = self._world_.getOrgFromPos(self.getRow(), self.getCol() + 1)
                        if (other.getStrength() < self.getStrength()):
                            self.move(self.getRow() , self.getCol() + 1)
                    else:
                        self.move(self.getRow() , self.getCol() + 1)
        elif rand == 2: #dol
            if (self._world_.isInsideWorld(self.getRow() + 1, self.getCol())):
                    if (self._world_.getOrgFromPos(self.getRow() + 1, self.getCol()) != None):
                        other = self._world_.getOrgFromPos(self.getRow() + 1, self.getCol())
                        if (other.getStrength() < self.getStrength()):
                            self.move(self.getRow() + 1, self.getCol())
                    else:
                        self.move(self.getRow() + 1, self.getCol())
        elif rand == 3: # lewo
            if (self._world_.isInsideWorld(self.getRow(), self.getCol() - 1)):
                    if (self._world_.getOrgFromPos(self.getRow(), self.getCol() - 1) != None):
                        other = self._world_.getOrgFromPos(self.getRow(), self.getCol() - 1)
                        if (other.getStrength() < self.getStrength()):
                            self.move(self.getRow(), self.getCol() - 1)
                    else:
                        self.move(self.getRow(), self.getCol() - 1)
        else:
            pass